### PR TITLE
APPS-928 Add instanceTypeSelection to task & workflow annotations.

### PR DESCRIPTION
**Pull request by r-i-v-a**

### CODE REVIEW

**Comment by r-i-v-a** [comment]
PR https://github.com/dnanexus/dxCompiler/pull/236

Once this is merged, I will be able to start a release of dxCompiler with initial support for creating global WDL workflows, which can be used as the comparison version in the APPS-624 dx-toolkit PR. @jdidion  @commandlinegirl

**Comment by r-i-v-a** [comment]
Waiting for current integration tests run to finish, but previous runs have succeeded
-- https://github.com/dnanexus/dxCompiler/actions/runs/1490337969
-- https://github.com/dnanexus/dxCompiler/actions/runs/1490367072

File: compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala [hunk 1]
```diff
@@ -370,7 +370,17 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
       case Some(JsString(s)) => s"${listMd("Hard-coded instance type")}${listMd2(s)}"
       case _                 => ""
     }
-    val md = s"${filesMd}${networkDockerImageMd}${dynamicDockerImageMd}${staticInstanceTypeMd}"
+    val staticInstanceTypeSelectionMd = instanceTypeSelection match {
+      case InstanceTypeSelection.Static =>
+        s"${listMd(
```
Comment (jdidion) [inline compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala]: `listMd` returns a string - why put it inside an interpolation?

File: compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala [hunk 2]
```diff
@@ -262,7 +263,16 @@ case class WorkflowCompiler(separateOutputs: Boolean,
           case Some(JsString(s)) => s"${listMd("Hard-coded instance type")}${listMd2(s)}"
           case _                 => ""
         }
-        s"${filesMd}${networkDockerImageMd}${dynamicDockerImageMd}${staticInstanceTypeMd}"
+        val staticInstanceTypeSelectionMd = instanceTypeSelection match {
+          case InstanceTypeSelection.Static =>
+            s"${listMd(
+                "Available instance types determined during WDL compilation will be " +
```
Comment (jdidion) [inline compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala:269]: same here

**Review by jdidion — CHANGES_REQUESTED** [review]

**Review by jdidion — APPROVED** [review]

**Pull request merged**